test/e2e/e2e_test.go

```diff
@@ -14,7 +14,7 @@
 
 // Modified from https://github.com/kubernetes-sigs/kubebuilder/tree/39224f0/test/e2e/v3
 
-package e2e
+package e2e_test
 
 import (
 	"testing"
```

Comment:
Replace:

``` 
func TestE2E(t *testing.T) {
	RegisterFailHandler(Fail)
	RunSpecs(t, "Operator SDK e2e suite")
}
```

With: 

```
// TestE2EGo ensures the Helm projects built with the Go tool by using its binary.
func TestE2EHelm(t *testing.T) {
	if testing.Short() {
		t.Skip("skipping Operator SDK E2E Go Suite testing in short mode")
	}
	RegisterFailHandler(Fail)
	RunSpecs(t, "E2EGo Suite")
}
```

Otherwise the test will be called in the target `make test-unit`

Reply:
`make test-unit` skips the entire test package with `go list ./... | grep -v -E 'github.com/operator-framework/operator-sdk/test/'`.

Reply:
HI @estroz and @joelanford, 

IMO we must implement solutions idempotent for the tests which means that they should not rely on the Travis or make targets. So, if a developer run `go tests ../... ` in the project it should call the tests that should be executed and finish with success what will not occur in this case.   

However, it is addressed in the PR: https://github.com/operator-framework/operator-sdk/pull/3499/files

Reply:
Otherwise, if we do not force as to follow up this kind of path then it contributes to the scenario where is not possible run the tests outside of the cluster.